src/tests/conformance/post_handshake/msg_of_interest.rs

```diff
@@ -39,5 +43,62 @@ async fn c005_MSG_OF_INTEREST_expect_after_connect() {
     node.stop().expect("unable to stop the node");
 }
 
-// TODO(Rqnsom): c005_t2: when the node initiates the connection, send MSG_OF_INTEREST with all messages enabled
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn c005_t2_MSG_OF_INTEREST_send_after_connect() {
+    // ZG-CONFORMANCE-005
+
+    // Spin up a node instance.
+    let target = TempDir::new().expect("couldn't create a temporary directory");
+    let mut node = Node::builder()
+        .build(target.path())
+        .expect("unable to build the node");
+    node.start().await;
+
+    // Create a synthetic node and enable handshaking.
+    let mut synthetic_node = SyntheticNodeBuilder::default()
+        .build()
+        .await
+        .expect("unable to build a synthetic node");
+
+    let net_addr = node.net_addr().expect("network address not found");
+
+    // Connect to the node and initiate the handshake.
+    synthetic_node
+        .connect(net_addr)
+        .await
+        .expect("unable to connect");
+
+    // Send a MsgOfInterest message with all expected tags included.
+    let tags = HashSet::from([
+        Tag::ProposalPayload,
+        Tag::AgreementVote,
+        Tag::MsgOfInterest,
+        Tag::MsgDigestSkip,
+        Tag::NetPrioResponse,
+        Tag::Ping,
+        Tag::PingReply,
+        Tag::ProposalPayload,
+        Tag::StateProofSig,
+        Tag::TopicMsgResp,
+        Tag::Txn,
+        Tag::UniEnsBlockReq,
+        Tag::VoteBundle,
+    ]);
+    let message = Payload::MsgOfInterest(MsgOfInterest { tags });
+    assert!(synthetic_node.unicast(net_addr, message).is_ok());
+
+    // Clear any remaining messages in the inbound queue.
+    let any_msg = Duration::from_millis(10);
```

Comment:
This has confusing var name. Also, I don't get what's the logic here: so we wait 10ms, then clear all messages, then wait for any message?

Reply:
I tried to use the short name to prevent fmt from breaking the line below into multiple lines.
The name `any_msg` is a short version of: `timeout we wait for any messages to arrive`
 `ten_mill` or `any_msg_to`?

Or what would you suggest?

Reply:
> so we wait 10ms, then clear all messages, then wait for any message?

Yeah, sort of - we wait until we receive nothing within the ten millis (that means the inbound_queue has been cleared and is empty).

The main logic here is:
After the node processes our `MsgOfInterest`, check that we are still receiving normal (any) messages as expected

Reply:
> I tried to use the short name to prevent fmt from breaking the line below into multiple lines. The name `any_msg` is > Or what would you suggest?
Since we use it only in one place, maybe it's more readable w/o var?

Reply:
Ah yes, but it will break it into multiple lines, but that's also fine
Consider it done!

Reply:
> check that we are still receiving normal (any) messages as expected
Maybe this should be more explicit, like wait for some other message that is not `MsgOfInterest`?